### DISCUSSION TITLE
Import psycopg2 only errors when running uvicorn

### DISCUSSION THREAD

**Discussion by lmason2** [fastapi/fastapi#10555] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import Depends, FastAPI
from typing_extensions import Annotated
from fastapi.requests import Request
from Factories.route_factory import RouteFactory
from pydantic import BaseModel, Base64Bytes
from Interfaces.postgres_connection import PostgresClient
import psycopg2

app = FastAPI()

@app.get('/')
async def root():
    return {'message': 'hello world'}

class RequestModel(BaseModel):
    base64_bytes: Base64Bytes

async def parse_body(request: Request):
    data: bytes = await request.body()
    return data

ParsingDependency = Annotated[str, Depends(parse_body)]


@app.post('/api/backyard/v1/{route_endpoint}')
async def handle_route(route_endpoint, request_body: ParsingDependency):
    route_handler = RouteFactory.get_handler(route_endpoint, request_body)
    return route_handler.results

@app.on_event('startup')
async def startup():
    # app.state.psql_client = PostgresClient('db_name', 'username', 'password')
    pass

if __name__ == '__main__':
    from Interfaces.postgres_connection import PostgresClient
    psql_client = PostgresClient('db_name', 'username', 'password')
    rows = psql_client.execute_postgres("SELECT * FROM table;")
    for row in rows:
        print(row)
```

### Description

When I run the above file with `python main.py`, the connection to my postgres client works without an issue.
When I run `uvicorn main:app --reload`, I get _ModuleNotFoundError: No module named 'psycopg2'_
I'm working in a venv, with the following pip list output:
Package            Version
------------------ ---------
annotated-types    0.5.0
anyio              3.7.1
certifi            2023.7.22
charset-normalizer 3.3.1
exceptiongroup     1.1.3
fastapi            0.103.2
idna               3.4
importlib-metadata 6.7.0
pip                22.0.4
psycopg2           2.9.9
psycopg2-binary    2.9.9
pydantic           2.4.2
pydantic_core      2.10.1
requests           2.31.0
setuptools         47.1.0
sniffio            1.3.0
starlette          0.27.0
typing_extensions  4.7.1
urllib3            2.0.7
zipp               3.15.0

### Operating System

macOS

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

Python 3.7.13

**Comment by jgould22** [comment, marked as answer]
Looks like uvicorn is install in your systems environment 

Try and install uvicorn (activate your venv and then `pip install 'uvicorn[standard]'`) into your virtual environment.

**Reply by lmason2** [reply]
Thank you!
